R/calcRenShareTargets.R

```diff
@@ -0,0 +1,352 @@
+
+
+        # set target year for all countries in REMIND region to target year of country with maximum historic electricity production
+        targetSelect[countries,MaxCountryYear,targetType] <- 1
+      }
+    }
+
+    # select targets by multiplying interpolated share targets with target selector targetSelect
+    # (if targetSelect is 1, target is included in target for REMIND region.
+    # if targetSelect is NA, it is not included)
+    x_select <- x_intp * targetSelect
+
+    return(x_select)
+  }
+
+  ### main function ----
+
+
+  ## 1. get data and select target types to be considered ----
+
+  # get NewClimate energy share targets and conert from percent to share
+  x <- readSource("NewClimate", subtype = "RenShareTargets")
+  x <- collapseDim(x / 100)
+
+  # calaulate renewable share targets of specific countries from technology-specific targets
+
+  # For Japan, add renewable electricity share targets per technology to total renewable share
+  x["JPN", , "SE|Electricity|Renewable"] <- x["JPN", , "SE|Electricity|Solar"] + x["JPN", , "SE|Electricity|Wind"] +
+    x["JPN", , "SE|Electricity|Geothermal"] + x["JPN", , "SE|Electricity|Biomass"] +
+    x["JPN", , "SE|Electricity|Hydro"]
+
+  # For Thailand, interpret renewable share target on FE electricity consumption as renewable share target on SE electricity generation
+  x["THA", , "SE|Electricity|Renewable"] <- x["THA", , "FE|Electricity|Renewable"]
+
+  # filter target types to be implemented
+  x <- x[, , c("SE|Electricity|Renewable", "SE|Electricity|Non-Biomass Renewable", "SE|Electricity|Non-Fossil", "FE|Renewable")]
+  # take minimum energy share for renewable / non-fossil targets,
+  # assume that if there is a range countries will interpret the lower bound of the range as criterion to fulfill the target
+  x <- collapseDim(x[, , "Min"])
+
+
+  ## 2. get historic renewable shares and totals ----
+
+  # historic energy share in 2020
+  xHist <- getHistEnergyShare(getNames(x), AvgSeveralYears = T)
+  xHistShare <- xHist[[1]]
+  xHistTotal <- xHist[[2]]
+
+  ## 3. extrapolate targets to REMIND time steps ----
+
+  # extrapolate targets to REMIND time steps where targets are in years that are not REMIND timesteps
+  x <- extrapolateTargetYears(x, xHistShare, seq(2020, 2050, 5))
+
+  ## 4. make assumptions about target development over time for countries with and without target ----
+
+  # interpolate and extrapolate renewable share targets for countries with targets
+  # assume that countries with share targets reach their share linearly from 2020 shares to the target year
+  # assume that after target year, renewable share targets stay constant
+  # assume that countries without target keep their historic share in the future
+  x_intp <- x
+  x_intp[,"y2020",] <- xHistShare
+  x_intp <- as.quitte(x_intp) %>%
+              mutate( "variable" = .data$data) %>%
+              select(-.data$data) %>%
+              interpolate_missing_periods(expand.values = T) %>%
```

Comment:
FYI: You do not need to convert to quitte just to do interpolation. You could use `toolFillYears` as well to do it directly on the magclass object. However, as the NA years per time series vary in `x_intp`, you would still have to iterate over regions and third dimension (like here https://github.com/pik-piam/mrremind/blob/master/R/calcGAINS2025.R#L217-L224), so not sure if this is more elegant.